src/qcodes/instrument_drivers/Keysight/keysight_e4980a.py

```diff
@@ -371,14 +371,16 @@ def __init__(
         )
         """This parameter tracks the signal mode which is being set."""
 
-        self.add_submodule("_correction", KeysightE4980ACorrection(self, "correction"))
+        self._correction: KeysightE4980ACorrection = self.add_submodule(
+            "_correction", KeysightE4980ACorrection(self, "correction")
+        )
         self._set_signal_mode_on_driver_initialization()
         self.connect_message()
 
     @property
     def correction(self) -> KeysightE4980ACorrection:
-        submodule = self.submodules["_correction"]
-        return cast("KeysightE4980ACorrection", submodule)
+        submodule = self._correction
```

Comment:
Perhaps just drop this property and rename the module. That would change the snapshot but for the better